### PR TITLE
Fix tile reloading race condition

### LINKED ISSUE #2559: Fix DataCloneError in Firefox and IE11
On firefox we get:

> DataCloneError: The object could not be cloned.

This error is encountered reliably on Firefox 43-46, hitting 217 users and 1.1k times so far. Occurs within the Actor object.

### PR DESCRIPTION
**Pull request by mourner**
Attempt to fix #2559. After this change, I can’t seem to reproduce the original bug. 

Will try writing a test case for this but not sure if it’s easy to do.

cc @lucaswoj @anandthakker

### CODE REVIEW

**Comment by anandthakker** [comment]
👍 Fix LGTM

> Will try writing a test case for this but not sure if it’s easy to do.

Shouldn't be too hard by mocking the dispatcher?  (like in [GeoJSONSource test](https://github.com/mapbox/mapbox-gl-js/blob/master/test/js/source/geojson_source.test.js))

**Comment by mourner** [comment]
@lucaswoj 👀, now a bit simpler and with a test.

File: test/js/source/vector_tile_source.test.js [hunk 1]
```diff
@@ -112,6 +112,34 @@ test('VectorTileSource', function(t) {
...
+        source.on('load', function () {
+            var tile = {
+                coord: new TileCoord(10, 5, 5, 0),
+                state: 'loading',
+                loadVectorData: function () {
+                    this.state = 'loaded';
+                    events.push('tile loaded');
+                }
+            };
+            source.loadTile(tile, function () {});
```
Comment (anandthakker) [inline test/js/source/vector_tile_source.test.js]: Just to be extra explicit, how about asserting `t.same(tile.state, 'loading')` after the first `loadTile` call?  The `setTimeout` in dispatcher.send ensures this, but I think including the assertion would make the test code a better expression of what we're trying to cover here

Comment (mourner) [inline test/js/source/vector_tile_source.test.js]: Good point, added.

**Comment by lucaswoj** [comment]
This looks great! Let's 🚢 when CI passes.

**Pull request merged**